app/presenters/icims/candidate_retry_imports_presenter.rb

```
@@ -0,0 +1,38 @@
+module Icims
+  class CandidateRetryImportsPresenter
+    DEFAULT_ERROR_MESSAGE = 'Check your email for details on errors'
+
+    def initialize(candidate, imported_result)
+      @candidate = candidate
+      @imported_result = imported_result
+    end
+
+    def successful_import?
+      if imported_result
+        imported_result.success?
+      else
+        false
+      end
+    end
+
+    def candidate_name
+      if candidate
+        candidate.name
+      else
+        ''
+      end
+    end
+
+    def import_error
+      if imported_result
+        imported_result.error
+      else
+        DEFAULT_ERROR_MESSAGE
+      end
+    end
+
+    private
+
+    attr_reader :imported_result, :candidate
+  end
```

@viola yes, they can be nil. But I'm not sure if this is the correct implementation I feel that I'm hiding the actual error which is what I'm investing now after your comment. Thanks ;-)